### PR TITLE
Add News & Communications and Transparency content

### PR DESCRIPTION
**Pull request by MahmudH**
- Retrieves most recent content for News and Communications supergroup
- Retrieves most recent content for Transparency supergroup content

### CODE REVIEW

File: app/services/most_recent_content.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+require 'gds_api/rummager'
+
+class MostRecentContent
+  include RummagerFields
+
+  attr_reader :content_ids, :filter_content_purpose_supergroup, :number_of_links
+
+  def initialize(content_ids:, filter_content_purpose_supergroup:, number_of_links: 5)
```
Comment (oscarwyatt) [inline app/services/most_recent_content.rb]: As it's being displayed in an element that shows three to a row, I imagine at some point we might want to change this to being a multiple of 3? Or we remove the default and make it an explicit choice every time it's displayed?

File: app/presenters/supergroups/news_and_communications.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
+module Supergroups
+  class NewsAndCommunications
+    attr_reader :content
```
Comment (vanitabarrett) [inline app/presenters/supergroups/news_and_communications.rb]: Not sure if we need this?

File: app/presenters/supergroups/transparency.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
+module Supergroups
+  class Transparency
+    attr_reader :content
```
Comment (vanitabarrett) [inline app/presenters/supergroups/transparency.rb]: Not sure if we need this?

File: app/services/most_recent_content.rb (hunk 1 above)
Comment (vanitabarrett) [reply app/services/most_recent_content.rb]: 5 is ok for now, as we will be showing the 5 most recent documents across 2 components - 3 in the highlight boxes and 2 in the document list. But we retrieve the 5 docs in one call to rummager 🙂

File: app/controllers/content_items_controller.rb [hunk 4]
```diff
@@ -69,9 +69,13 @@ def load_taxonomy_navigation
 
       taxon_ids = taxons.map { |taxon| taxon["content_id"] }
       services = Supergroups::Services.new(taxon_ids)
+      news_and_communications = Supergroups::NewsAndCommunications.new(taxon_ids)
+      transparency = Supergroups::Transparency.new(taxon_ids)
 
       @taxonomy_navigation = {
         services: (services.all_services if services.any_services?),
+        news_and_communications: news_and_communications.tagged_content,
```
Comment (vanitabarrett) [inline app/controllers/content_items_controller.rb]: We need a method here (and for transparency) that's similar to the one above for services, that first checks to see if there are any services before adding them to the hash.

File: app/presenters/supergroups/news_and_communications.rb (hunk 2 above)
Comment (MahmudH) [reply app/presenters/supergroups/news_and_communications.rb]: 👍

File: app/presenters/supergroups/news_and_communications.rb [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    def initialize(taxon_ids)
+      @taxon_ids = taxon_ids
+      @content = MostRecentContent.fetch(content_ids: @taxon_ids, filter_content_purpose_supergroup: "news_and_communications")
+    end
+
+    def tagged_content
+      format_document_data(@content)
+    end
+
+    def any_services?
```
[the remark this answers is not in the document]
Comment (MahmudH) [reply app/presenters/supergroups/news_and_communications.rb]: Do you think I should have any_content? instead or be more specific and do any_news?

[the remark this answers is not in the document]
Comment (vanitabarrett) [reply app/presenters/supergroups/news_and_communications.rb]: `any_content?` would probably be good 👍

[the remark this answers is not in the document]
Comment (MahmudH) [reply app/presenters/supergroups/news_and_communications.rb]: 👍

File: app/presenters/supergroups/transparency.rb [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    def initialize(taxon_ids)
+      @taxon_ids = taxon_ids
+      @content = MostRecentContent.fetch(content_ids: @taxon_ids, filter_content_purpose_supergroup: "transparency")
+    end
+
+    def tagged_content
+      format_document_data(@content)
+    end
+
+    def any_services?
```
[the remark this answers is not in the document]
Comment (MahmudH) [reply app/presenters/supergroups/transparency.rb]: 👍

**Comment by sihugh** [comment]
Closing this as it's been superseded by #1002 (sorry!)

**Pull request closed without merging**